api/v1alpha1/cre_condition_consts.go

```diff
@@ -19,42 +19,32 @@ limitations under the License.
 package v1alpha1
 
 const (
-	// PipelineRunCreated indicates that the Tekton pipeline run was created
-	PipelineRunCreated = "PipelineRunCreated"
 
-	// ErrorPipelineRunCreate indicates that the Tekton pipeline run creation failed
-	ErrorPipelineRunCreate = "ErrorPipelineRunCreate"
-
-	// ImportingImage indicates that the image is being imported from a remote registry
-	ImportingImage = "ImportingImage"
-
-	// RequredSecretMissing indicates that the secret required for authentication to the container image registry is missing
-	RequiredSecretMissing = "RequiredSecretMissing"
-
-	// ValidatingImportedImage indicates that the imported image is being validated by a Tekton PipelineRun's Step
-	ValidatingImportedImage = "ValidatingImportedImage"
-
-	// ImageImportReady indicates that the imported image is ready to be used
-	ImageImportReady = "ImageImportReady"
-
-	// ImageImportInvalid indicates that the imported image is invalid
-	ImageImportInvalid = "ImageImportInvalid"
-
-	// ErrorResolvingDependencies indicates that the dependency resolution failed during preparation of the image build
-	ErrorResolvingDependencies = "ErrorResolvingDependencies"
-
-	// BuildingImage indicates that the image is being built by a Tekton PipelineRun
-	BuildingImage = "BuildingImage"
-
-	// PackageListBuildCompleted indicates that the package list build completed
-	PackageListBuildCompleted = "PackageListBuildCompleted"
-
-	// ErrorBuildingImage indicates that the image build failed
-	ErrorBuildingImage = "ErrorBuildingImage"
-
-	// GenericPipelineError indicates that the pipeline failed with an error
-	GenericPipelineError = "GenericPipelineError"
-
-	// PipelineRunCompleted indicates that the Tekton pipeline run completed
-	PipelineRunCompleted = "PipelineRunCompleted"
+	// There is at least one ImageStreamTag available in the corresponding ImageStream.
+	Ready = "Ready"
+
+	// Reasons for (False -> True):
+	// - ImagePushed
+	// Reasons for (True -> False):
+	// - ImageStreamDeleted
+	// - ImageTagDeleted
+
+	// The ImageStreamTag corresponding to the current CRE observedGeneration exists.
+	ImageUpToDate = "ImageUpToDate"
+
+	// Reasons for (False -> True):
+	// - ImagePushed
+	// Reasons for (True -> False):
+	// - ImageStreamDeleted
+	// - ImageTagDeleted
+	// - NewCreVersion
+
+	// The last pipeline building the image errored out.
+	BuildFailing = "BuildFailing"
+
+	// Reasons for (False -> True):
+	// - PipelineSucceded
+	// Reasons for (True -> False):
+	// - PipelineFailed
```

Comment:
It's just personal preference, but I think using the affirmative is more readable here. We can also add the a new pipeline run as a reason for the status being unknown.

```suggestion
	LatestBuildSucceeded = "BuildSucceeded"

        // Reasons for (True -> Unknown || False -> Unknown)
        // - NewPipelineRun
	// Reasons for (Unknown -> True):
	// - PipelineRunSucceeded
	// Reasons for (Unknown -> False):
	// - PipelineRunFailed
```

Reply:
I think it is indeed better. `Failing` was to express that the current build (which presumably we might retry with exponential back-off like other k8s controllers do) was in a failure state. Adding 'Latest' does express that.